emma/OntoEmma.py

```diff
@@ -184,33 +182,49 @@ def load_alignment(self, gold_path):
                 )
 
     def train(
-        self, model_type: str, model_path: str, config_file: str
+        self, model_type: str, model_path: str, config_file: str, cuda_device: int = -1
     ):
         """
         Train model
         :param model_type: type of model (nn, lr etc)
         :param model_path: path to ontoemma model
         :param config_file: path to training data, dev data, config for nn
+        :param cuda_device: GPU device number
         :return:
         """
         assert model_type in constants.IMPLEMENTED_MODEL_TYPES
         assert config_file is not None
         assert os.path.exists(config_file)
 
+        # train NN model with AllenNLP
```

Comment:
I completely agree. I actually already refactored in this way for train/evaluate/align in another branch, so I will probably open another PR with it instead of trying to apply it to this branch.